pom.xml

```diff
@@ -52,6 +52,11 @@
       <version>5.4.2</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>uk.com.robust-it</groupId>
+      <artifactId>cloning</artifactId>
+      <version>1.9.11</version>
+    </dependency>
```

Comment:
What is this dependency used for?

Reply:
It is used to make a `Request` copy (so it will be added as the first part of transaction with no position)

Reply:
I looked at this dependency and I don't think it is appropriate. There are many issues filed that indicate the library does not work with Java 10+ unless you start Java with an argument to allow reflection to work outside the module barriers put in place in the latest Java. We will likely be going to Java 11 sometime next year ([FOLIO-1772](https://issues.folio.org/browse/FOLIO-1772)). Perhaps they will solve this problem by that time, but, given their current lack of attention to the existing issues highlighting the failure, it does not seem likely.

Reply:
Is the [Apache Commons SerializationUtils.clone](http://commons.apache.org/proper/commons-lang/javadocs/api-3.9/org/apache/commons/lang3/SerializationUtils.html) method appropriate?

Reply:
[This article](https://www.baeldung.com/java-deep-copy) suggested doing a roundtrip using Jackson to achieve a deep copy.

If we decide to continue down this route, I suggest we use that mechanism for the time being. 

I am aware it is not ideal, as it requires handling checked exceptions and likely has a performance impact.

A possible implementation provided below.

```
    final ObjectMapper mapper = ObjectMapperTool.getMapper();

    return mapper.readValue(mapper.writeValueAsString(requestToCopy), Request.class);
```

Reply:
I think it is a common issue with reflection API since Java 9. It should affect even Mockito and PowerMockito libs. Anyway, I've chosen to use SQL query to remove positions instead of copy.

Reply:
The difference is those are test frameworks, where we could loosen restrictions if need be. This was in production code. Anyway, the dependency is no longer needed, so I am fine. 😄